src/main/java/org/opentripplanner/framework/geometry/WorldEnvelope.java

```diff
@@ -68,6 +61,14 @@ public double getUpperRightLatitude() {
     return upperRightLatitude;
   }
 
+  public double calcCenterLatitude() {
+    return (getLowerLeftLatitude() + getUpperRightLatitude()) / 2;
+  }
+
+  public double calcCenterLongitude() {
+    return (getLowerLeftLongitude() + getUpperRightLongitude()) / 2;
```

Comment:
I don't think this is correct, if we cross the 180th meridian, as is described in the javadoc

Reply:
Yes I think you are correct. This should be encapsulated and unit tested. I did not change the code, only moved it out of the graph. The code on line 113 and 114 in the WorldEnvelope look wrong as well - I would expect left and right to flip if we are crossing 180 degrees. The envelope is used in several places and it varies how it is calculated. In addition to this there is similar calculations in the StopService. This is a good candidate for making an encapsulated "micro-service" witch if requested(injected anywhere) should calculate the envelope and center and provide that to all clients. I will add a unit test, in a different pr.

Reply:
I found more issues with the `WorldEnvelope` after writing a unit test one it. I will refactor it and create a new small PR. It fails not only for coordinates on both side of 180 degrees (witch we could accept), but it also fails when there is 1 coordinate on the other side of 0 degrees.